tests/ui/map_clone.rs

```diff
@@ -97,4 +97,7 @@ fn map_clone_deref() {
     let _: Option<i32> = x.as_ref().map(|y| **y);
 }
 
-fn main() { }
+fn main() {
+    // used to be a false positive
```

Comment:
Midnight idea: This could go into a `fn former_false_positive() { .. }` function for extra self-documenting test code. I'm OK with it either way, though.